crates/xxi-node/src/message_handler.rs

```diff
@@ -658,10 +706,102 @@ fn read_tentenone_message<R: ::std::io::Read>(
+    use crate::message_handler::TenTenOneSettleOffer;
+    use crate::message_handler::WireMessage;
+    use crate::node::event::NodeEventHandler;
+    use dlc_manager::DlcChannelId;
+    use dlc_messages::channel::SettleOffer;
+    use lightning::ln::wire::CustomMessageReader;
+    use lightning::ln::wire::Type;
+    use lightning::util::ser::Readable;
+    use lightning::util::ser::Writeable;
+    use secp256k1::PublicKey;
+    use std::io::Cursor;
+    use std::str::FromStr;
+    use std::sync::Arc;
+    use time::OffsetDateTime;
+
+    #[test]
+    fn test_impl_serde_writeable() {
+        let order = commons::Order {
+            id: Default::default(),
+            price: Default::default(),
+            leverage: 0.0,
+            contract_symbol: ContractSymbol::BtcUsd,
+            trader_id: PublicKey::from_str(
+                "02d5aa8fce495f6301b466594af056a46104dcdc6d735ec4793aa43108854cbd4a",
+            )
+            .unwrap(),
+            direction: Direction::Long,
+            quantity: Default::default(),
+            order_type: OrderType::Market,
+            timestamp: OffsetDateTime::now_utc(),
+            expiry: OffsetDateTime::now_utc(),
+            order_state: OrderState::Open,
+            order_reason: OrderReason::Manual,
+            stable: false,
+        };
+
+        let settle_offer = TenTenOneMessage::SettleOffer(TenTenOneSettleOffer {
+            settle_offer: SettleOffer {
+                channel_id: DlcChannelId::default(),
+                counter_payout: 0,
+                next_per_update_point: secp256k1_zkp::PublicKey::from_str(
+                    "02d5aa8fce495f6301b466594af056a46104dcdc6d735ec4793aa43108854cbd4a",
+                )
+                .unwrap(),
+                timestamp: 0,
+                reference_id: None,
+            },
+            order,
+            filled_with: FilledWith {
+                order_id: Default::default(),
+                expiry_timestamp: OffsetDateTime::now_utc(),
+                oracle_pk: secp256k1::XOnlyPublicKey::from_str(
+                    "cc8a4bc64d897bddc5fbc2f670f7a8ba0b386779106cf1223c6fc5d7cd6fc115",
+                )
+                .unwrap(),
+                matches: vec![],
+            },
+        });
+
+        let mut buf = Vec::new();
+        settle_offer.type_id().write(&mut buf).unwrap();
+        settle_offer.write(&mut buf).unwrap();
+
+        let mut reader = Cursor::new(&mut buf);
+        let message_type =
+            <u16 as Readable>::read(&mut reader).expect("to be able to read the type prefix.");
+
+        let handler = TenTenOneMessageHandler::new(Arc::new(NodeEventHandler::new()));
+        let message = handler
+            .read(message_type, &mut reader)
+            .expect("to be able to read the message")
+            .expect("to have a message");
+
+        assert!(matches!(
+            message,
+            WireMessage::Message(TenTenOneMessage::SettleOffer(_))
```

Comment:
🔧 I think we should check against the whole body. It would be a good opportunity to use `insta` too.

Reply:
`insta` is really cool. thanks for the tip.